llvm/lib/CodeGen/StackProtector.cpp

```diff
@@ -297,7 +298,13 @@ bool StackProtector::RequiresStackProtector() {
           continue;
         }
 
-        if (Strong && HasAddressTaken(AI)) {
+        /// VisitedPHIs - The set of PHI nodes visited when determining
+        /// if a variable's reference has been taken.  This set
+        /// is maintained to ensure we don't visit the same PHI node multiple
+        /// times.
+        SmallPtrSet<const PHINode *, 16> VisitedPHIs;
```

Comment:
I don't think this is equivalent--it would want to be declared outside the `for (const BasicBlock &BB : *F) {` loop up at line 247.

Reply:
I've updated the patch to fix this.